### PR TITLE
Af 146 configargparse

### PR DESCRIPTION
**Pull request by afaulconbridge**
Mostly solves issue opentargets/platform#146

This is a big PR so may need some discussion face-to-face or followup issues to refine. It intends to solve the easy 90% of the configuration, but the remaining 10% will be much easier to do if we wait until other issues have been resolved e.g. replacing Redis queues with pypeln opentargets/platform#361

I'm putting the PR in now because the changes are becoming too big to maintain outside of master branch, even though it is not a complete issue. The changes are more than sufficient to resolve the underling problems (reproducibility, consistency, etc).

### CODE REVIEW

File: mrtarget/modules/QC.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def run_associationQC(self):
     def run_evidence2associationQC(self):
         computed_assocations_ids = set(self.esquery.get_all_associations_ids())
         missing_assocations_ids = set()
-        total_evidence = self.esquery.count_elements_in_index(Config.ELASTICSEARCH_DATA_INDEX_NAME+'*')
+        total_evidence = self.esquery.count_elements_in_index(Const.ELASTICSEARCH_DATA_INDEX_NAME+'*')
```
Comment (mkarmona) [inline mrtarget/modules/QC.py]: the `*` can be removed if I am not wrong with it.

File: mrtarget.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+#TODO move this to GCP https://github.com/opentargets/platform/issues/379
+mouse-phenotypes-orthology: http://www.informatics.jax.org/downloads/reports/HMD_HumanPhenotype.rpt
+#TODO move this to GCP https://github.com/opentargets/platform/issues/379
+mouse-phenotypes-report: http://www.informatics.jax.org/downloads/reports/MGI_PhenoGenoMP.rpt
+
+#cancer biomarkers plugin
+biomarker: https://storage.googleapis.com/ot-releases/18.12/annotations/cgi_biomarkers_per_variant.tsv
+
+#chemical probes plugin
+chemical-probes-1: https://storage.googleapis.com/ot-releases/18.12/annotations/chemicalprobes_portalprobes_20181130.tsv
```
Comment (mkarmona) [inline mrtarget.yml]: it is not `-1 and -2` those are two different files which it has to be
- chemical-probes-portalprobes
- chemical-probes-probeminer

File: mrtarget.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+#uniprot
+uniprot-uri: https://storage.googleapis.com/ot-releases/18.12/annotations/uniprot-20181130.xml.gz
+
+#gene plugins
+gene-data-plugin-places: [mrtarget/plugins/gene]
+gene-data-plugin-names: [HGNC,Orthologs,Ensembl,Uniprot,ChEMBL,MousePhenotypes,Hallmarks,CancerBiomarkers,ChemicalProbes,Tractability]
+
+#hgnc gene plugin
+#TODO move this to GCP https://github.com/opentargets/platform/issues/326
+hgnc-complete-set: http://ftp.ebi.ac.uk/pub/databases/genenames/new/json/hgnc_complete_set.json
```
Comment (MichaelaEBI) [inline mrtarget.yml]: Will this be downloaded and added to the https://storage.googleapis.com/open-targets-data-releases/_releaseTag_/input/annotation-files folder?

File: mrtarget.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+#ortholog gene plugin
+#TODO move this to GCP https://github.com/opentargets/platform/issues/326
+hgnc-orthologs: http://ftp.ebi.ac.uk/pub/databases/genenames/hcop/human_all_hcop_sixteen_column.txt.gz
+hgnc-orthologs-species: [9606-human,9598-chimpanzee,9544-macaque,10090-mouse,10116-rat,9615-dog,9823-pig,8364-frog,7955-zebrafish,7227-fly,6239-worm,4932-yeast]
+
+#mouse phenotype plugin
+#TODO move this to GCP https://github.com/opentargets/platform/issues/379
+mouse-phenotypes-orthology: http://www.informatics.jax.org/downloads/reports/HMD_HumanPhenotype.rpt
+#TODO move this to GCP https://github.com/opentargets/platform/issues/379
+mouse-phenotypes-report: http://www.informatics.jax.org/downloads/reports/MGI_PhenoGenoMP.rpt
```
Comment (MichaelaEBI) [inline mrtarget.yml]: The same for ortholog and mouse phenotype files - will they be added to the annotations folder?

**Comment by afaulconbridge** [comment]
After discussion, this will have more work done. We decided to split the config into "execution config" and "data config". The execution config is installation specific - hostnames, number of workers, index prefix, logging, etc. The data config is release specific - validation schema, evidence files, annotation files, etc.

This PR will be closed, and I'll file a new one with those improvements.

File: mrtarget/Settings.py [hunk 5]
```diff
@@ -99,226 +89,15 @@ def read_option(option, cast=None, ini=ini, section='dev',
...
-    EVIDENCEVALIDATION_PROVIDER_EMAILS["cttv011"] = [ 'eddturner@ebi.ac.uk', 'bpalka@ebi.ac.uk' ]
-    EVIDENCEVALIDATION_PROVIDER_EMAILS["cttv012"] = [ 'tsmith@ebi.ac.uk', 'garys@ebi.ac.uk', 'cyenyxe@ebi.ac.uk' ]
-    EVIDENCEVALIDATION_PROVIDER_EMAILS["cttv025"] = [ 'kafkas@ebi.ac.uk', 'ftalo@ebi.ac.uk' ]
-    EVIDENCEVALIDATION_FILENAME_REGEX = re.compile(r"""
-    (?P<datasource>[a-zA-Z0-9_]+)(\-
-    (?P<d1>\d{2})\-
-    (?P<d2>\d{2})\-
-    (?P<d3>\d{4}))?
-    (?P<suffix>\.json\.gz|\.json|\.json\.zip)$""", re.VERBOSE)
+    RELEASE_VERSION = read_option('CTTV_DATA_VERSION', default='')
```
Comment (MichaelaEBI) [inline mrtarget/Settings.py]: CTTV?

Comment (afaulconbridge) [reply mrtarget/Settings.py]: Legacy compatibility!

File: mrtarget.yml (hunk 3 above)
Comment (afaulconbridge) [reply mrtarget.yml]: Yes, see issue opentargets/platform#326

File: mrtarget.yml (hunk 2 above)
Comment (afaulconbridge) [reply mrtarget.yml]: The numeric suffix is how they are currently configured in the platform. Changing them is outside the scope of this issue, it should be a separate branch & PR

**Pull request closed without merging**